tests/networkpolicy_test.go

```diff
@@ -0,0 +1,111 @@
+		ktests.SkipIfNotUseNetworkPolicy(virtClient)
+		ktests.BeforeTestCleanup()
+	})
+
+	Context("Networkpolicy allow http port", func() {
+		It("[test_id:CNV-369] Set connectivity to only allow from http port", func() {
+
+			By("Create VMIs")
+			userData := "#cloud-config\npassword: fedora\nchpasswd: { expire: False }\n"
+			vmia = ktests.NewRandomVMIWithEphemeralDiskAndUserdata(ktests.ContainerDiskFor(ktests.ContainerDiskFedora), userData)
+			ktests.AddExplicitPodNetworkInterface(vmia)
+
+			vmia.Spec.Domain.Resources.Requests[k8sv1.ResourceName("memory")] = resource.MustParse("1024M")
+			_, err = virtClient.VirtualMachineInstance(ktests.NamespaceTestDefault).Create(vmia)
+			Expect(err).ToNot(HaveOccurred())
+
+			vmib = ktests.NewRandomVMIWithEphemeralDiskAndUserdata(ktests.ContainerDiskFor(ktests.ContainerDiskCirros), "#!/bin/bash\necho 'hello'\n")
+			_, err = virtClient.VirtualMachineInstance(ktests.NamespaceTestDefault).Create(vmib)
+			Expect(err).ToNot(HaveOccurred())
+
+			ktests.WaitForSuccessfulVMIStart(vmia)
+			vmia, err = virtClient.VirtualMachineInstance(ktests.NamespaceTestDefault).Get(vmia.Name, &v13.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Install httpd and start the service inside the VMI")
+			ktests.WaitUntilVMIReady(vmia, ktests.LoggedInFedoraExpecter)
+			err = ktests.CheckForTextExpecter(vmia, []expect.Batcher{
+				&expect.BSnd{S: "\n"},
+				&expect.BExp{R: "#"},
+				&expect.BSnd{S: "sudo yum install httpd -y\n"},
+				&expect.BExp{R: "#"},
+				&expect.BSnd{S: "echo hellokubevirt | sudo tee /var/www/html/index.html\n"},
+				&expect.BExp{R: "hellokubevirt"},
+				&expect.BSnd{S: "sudo systemctl start httpd\n"},
+				&expect.BExp{R: "#"},
+				&expect.BSnd{S: "curl --silent --connect-timeout 5 localhost:80 | grep hellokubevirt | wc -l \n"},
+				&expect.BExp{R: "1"},
+			}, 180)
+			Expect(err).ToNot(HaveOccurred())
+
+
+			By("Create deny-all networkpolicy and validate all the traffic are limited")
+			tests.CreateResourceWithFilePath(denyallYaml)
+			ip := vmia.Status.Interfaces[0].IP
+			assertCurlFail(ip, vmib)
+
+			By("Create allow-tcp networkpolicy and validate http port is available")
+			tests.CreateResourceWithFilePath(allowhttpYaml)
+			assertCurlSucceed(ip, vmib)
+		})
+
+	})
+})
+
+
+func assertCurlFail(ip string, vmi *v1.VirtualMachineInstance) {
+	ktests.WaitUntilVMIReady(vmi, ktests.LoggedInCirrosExpecter)
+	err := ktests.CheckForTextExpecter(vmi, []expect.Batcher{
+		&expect.BSnd{S: fmt.Sprintf("curl --silent --connect-timeout 5 %s:80 | grep hellokubevirt | wc -l \n", ip)},
+		&expect.BExp{R: "0"},
+	}, 60)
+	Expect(err).ToNot(HaveOccurred())
+}
+
+
+func assertCurlSucceed(ip string, vmi *v1.VirtualMachineInstance) {
+	ktests.WaitUntilVMIReady(vmi, ktests.LoggedInCirrosExpecter)
+	err := ktests.CheckForTextExpecter(vmi, []expect.Batcher{
+		&expect.BSnd{S: fmt.Sprintf("curl --silent --connect-timeout 5 %s:80 | grep hellokubevirt | wc -l \n", ip)},
+		&expect.BExp{R: "1"},
```

Comment:
Please make only one function that check curl and the function will get the expected result that you need.
expect.BExp{R: "1"},

these two functions seem duplicated.

Reply:
Already updated. Thanks